asmopt/tests/optimizer.rs

```diff
@@ -0,0 +1,311 @@
+use powdr_analysis::analyze;
+use powdr_asmopt::optimize;
+use powdr_parser::parse_asm;
+
```

Comment:
test suggestion: machines in `link`s (outside of instructions) should not be removed

Reply:
Added, thanks!